sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -898,7 +903,15 @@ private void validateReplicaAddresses(AddressInformation[] addresses) {
                 Arrays
                     .stream(addresses)
                     .map(address -> address.getPhysicalUri())
-                    .filter(addressUri -> addressUri.getHealthStatus() == Uri.HealthStatus.UnhealthyPending)
+                    .filter(addressUri -> this.replicaValidationScopes.contains(addressUri.getHealthStatus()))
+                    .sorted(new Comparator<Uri>() {
```

Comment:
Just my 2 cents: I am not a fan of these stream-based or Linq-based operations liek sorted where it is unclear how the functional requirement is implemented.

Does sorted force using an ordered data structure? or use an unstructured data structure and then apply an order by after all results have been cached - like you would do when sorting the List you get back without this operator.

Perf-wise the latter would probably be better in this case>

Not blocking - and I don't have time to look into what .sorted actually is doing - just raising this as a general issue I have when these operators obfuscate the non-functional side-effects.